src/providers/doppler.js

```diff
@@ -0,0 +1,38 @@
+const https = require("https");
+
+/**
+ * Fetch secrets from Doppler the API.
+ * Requires the `DOPPLER_TOKEN` environment variable to be set. See https://docs.doppler.com/docs/enclave-service-tokens
+ * @returns {() => Promise<Record<string, string>>}
+ */
+async function fetch() {
+  if (!process.env.DOPPLER_TOKEN) {
+    throw new Error("Doppler API Error: The 'DOPPLER_TOKEN' environment variable is required");
+  }
+
+  return new Promise(function (resolve, reject) {
```

Comment:
FYI, totally optional: A pattern I find helpful is to immediately branch off of a resolved promise so that return values are easier to handle.

```js
return Promise.resolve()
  .then(() => {
    return JSON.parse(secrets) // automatically resolves
    // OR
    throw new Error("foo") // automatically rejects
  })
```